### ISSUE TITLE
Relase csfml 2.2 (or 2.1.1) ?

### ISSUE THREAD

**Issue by eleweek** [LaurentGomila/CSFML#50]
Hi, csfml 2.1 is incompatible with sfml 2.2 due to changes in blending mode stuff. This commit in masetr fixes the problem: https://github.com/LaurentGomila/CSFML/commit/d6849a3d57e41b8c9110c51686baae8fee876f82

I'm trying to update sfml 2.1 -> sfml 2.2 so it can build on yosemite with homebrew:

https://github.com/Homebrew/homebrew/pull/35279

However I'm having trouble with the csfml 2.1 <-> sfml 2.2 incompatibility. 

I think it would be nice to have compatibility between latest stable csfml and sfml releases. So I suggest making a stable release of csfml(if this is possible)

**Comment by LaurentGomila** [comment]
Do you really think that we were waiting for someone to open an issue, to release CSFML 2.2? :confused:

There's currently nobody to maintain CSFML (and SFML.Net). Any help will be appreciated though.

**Comment by eleweek** [comment]
> Do you really think that we were waiting for someone to open an issue, to release CSFML 2.2?

Well... yes? Seriously though, I assumed that maybe you weren't aware of this incompatibility because nobody bothered to report it. 

> There's currently nobody to maintain CSFML (and SFML.Net). Any help will be appreciated though.

I can't commit long-term, but if there are issues in code that need to be fixed, I'll be happy to help.

**Comment by LaurentGomila** [comment]
There's no "issue" to fix currently, the latest modifications to SFML just need to be ported.

**Comment by BlaXpirit** [comment]
CSFML is extremely important, because more than half of bindings to other languages depend on it (including the one I've just finished).

**Comment by eXpl0it3r** [comment]
The question you should be researching is: What needs to be updated before bumping the version number?
Besides the version number, CSFML seems to build fine with SFML 2.2 (or even SFML master).

**Comment by BlaXpirit** [comment]
upVector, requestFocus, sfSoundRecorder_setProcessingInterval, BlendMode, changes in fonts int->float, window focus, strikethrough   are what I found by scrolling through `git diff CSFML-2.1/include CSFML-master/include`.

**Comment by BlaXpirit** [comment]
So I've published the [changes in header files between [C]SFML 2.1 and 2.2](https://gist.github.com/BlaXpirit/0c9b87f1301c9cc3103d)

**Comment by LaurentGomila** [comment]
Very nice, thanks :smiley:

**Comment by zsbzsb** [comment]
I am working on finishing the last few changes.... but I dunno what you are talking about blending modes for. CSFML (and SFML.NET) have had those changes for a while now.

And my todo list...

CSFML

underline position/thickness
font info
joystick info
sound recorder start/bool return
sound recorder devices

-=SFML.NET=-

window focus has/request
underline position/thickness
font info
joystick info
sound recorder start/bool return
sound recorder devices

**Comment by LaurentGomila** [comment]
> color int constructor LaurentGomila/SFML#722

If you also apply modifications pending for 2.3, we won't be able to release a clean CSFML/SFML.Net 2.2. Make all the 2.2 updates, then we release the bindings, and after that you can start working on newer modifications.

**Comment by zsbzsb** [comment]
Alright, I was just adding it there so I didn't lose track of it.

**Comment by zsbzsb** [comment]
This can be closed once #53 is merged.

**Comment by BlaXpirit** [comment]
I see some unaddressed changes: Sensor, Touch, Http.
https://gist.github.com/BlaXpirit/0c9b87f1301c9cc3103d#file-sfml-diff

**Comment by zsbzsb** [comment]
https://github.com/LaurentGomila/CSFML/issues/50#issuecomment-73418731

So.... checklist is done. Verify it with @BlaXpirit header diff. https://gist.github.com/BlaXpirit/0c9b87f1301c9cc3103d#file-sfml-diff


Can we merge and then tag 2.2 now?  :smiling_imp:

**Comment by LaurentGomila** [comment]
Is everything ok for everyone else?

> Can we merge

Merge what?

**Comment by zsbzsb** [comment]
>Merge what?

https://github.com/LaurentGomila/CSFML/pull/53 #53

**Comment by BlaXpirit** [comment]
My bindings to another language work well. Tried some new functions, no problems at all.

**Comment by LaurentGomila** [comment]
Let's wait a few more days for last minute fixes. Bump me if I forget.

**Comment by zsbzsb** [comment]
>Let's wait a few more days for last minute fixes.

At first I was going to reply with sad face, but now I realized we need at least 2 more fixes. We need to add a function for relative touch position to a render window and now https://github.com/LaurentGomila/SFML/issues/806.

**Comment by zsbzsb** [comment]
Everything is done, so lets plan on doing the version change merge on Sunday (March 1st)?

**Comment by zsbzsb** [comment]
>Bump me if I forget.

:round_pushpin:

**Comment by zsbzsb** [comment]
@LaurentGomila this is now a formal bump. I propose that you have forgotten about this and as your friendly neighborhood programmer I have decided to remind you. CSFML 2.2 is complete so #53 can now be merged. :wink:

**Comment by LaurentGomila** [comment]
I haven't forgotten, I'm just very busy with real life problems. I should have left a word here, sorry.

**Comment by zsbzsb** [comment]
Well take your time, don't let me rush you or anything. If you have other stuff happening IRL that should take priority.

**Comment by eleweek** [comment]
Thank you very much! Could you please comment here when a tarball will be available on http://www.sfml-dev.org/download/csfml/ ?

**Comment by zsbzsb** [comment]
Working on builds now. Expect within a few days. :wink:

**Comment by eleweek** [comment]
:+1: Awesome, thank you!

**Comment by eXpl0it3r** [comment]
Till things are collected and I've more time to update everything, feel free to test these Windows build:

* [Windows x86](http://my-gate.net/pub/CSFML-2.2-windows-32-bit.zip)
* [Windows x64](http://my-gate.net/pub/CSFML-2.2-windows-64-bit.zip)

**Comment by mantognini** [comment]
OS X is here: http://1drv.ms/1Khhx0d

**Comment by zsbzsb** [comment]
Just until @eXpl0it3r can get all this on the website.

[Linux x86](http://hashcookie.net/uploads/9f44221d37_CSFML-2.2-linux-gcc-32bits.tar.bz2)
[Linux x64](http://hashcookie.net/uploads/bee1fcfa19_CSFML-2.2-linux-gcc-64bits.tar.bz2)

**Comment by LaurentGomila** [comment]
Windows builds are switched, 32 is 64 and 64 is 32.

**Comment by zsbzsb** [comment]
>And I don't see any dependency to a C runtime (msvcrt for VC++, or whatever for gcc), how were they compiled?

I asked @eXpl0it3r if he was going to compile them to use static std libs. Reason being because that is how they were done in the [previous releases](http://en.sfml-dev.org/forums/index.php?topic=12846.msg89912#msg89912). Also GCC was used if I am not mistaken.

**Comment by LaurentGomila** [comment]
The strange thing is that if you check the older releases, they all depend directly on MSVCRT.dll.

And sorry for the 32/64 bits thing, everything's correct, my bad.

**Comment by zsbzsb** [comment]
As far as I know MSVCRT is the *Microsoft Visual C Runtime* and is included in the base OS. Reason being is that it is a C library and the names aren't mangled so its not specific per compiler. So static libs to the MSVC compiler means static C++ runtime.

**Comment by LaurentGomila** [comment]
So they were compiled with gcc and static runtime? @eXpl0it3r?

I thought in this case they would be bigger, but they are even smaller than my previous builds with VC++ :neutral_face:

**Comment by eXpl0it3r** [comment]
I had like no time to build stuff, so I didn't test anything. Feel free to run a test with VS and GCC.

I compiled stuff with VS 2010 and link the runtime lib statically.

**Comment by zsbzsb** [comment]
I tested your builds @eXpl0it3r and they work just fine.

**Comment by eXpl0it3r** [comment]
With GCC as well?

**Comment by zsbzsb** [comment]
I dropped them into a SFML.NET project and to test them - didn't try linking. :/

**Comment by eXpl0it3r** [comment]
@mantognini Previously there was the description "Clang - universal 32+64-bit (OS X 10.8+, compatible with C++11 and libc++)" should I keep this?

**Comment by mantognini** [comment]
Good catch. It should now read "Clang - universal 32+64-bit (OS X 10.7+, compatible with C++11 and libc++)".

**Comment by eXpl0it3r** [comment]
http://www.sfml-dev.org/download/csfml/
http://en.sfml-dev.org/forums/index.php?topic=17680.0

**Comment by eleweek** [comment]
Yay! Thank you all!

**Comment by LaurentGomila** [comment]
Thank you very much @eXpl0it3r :clap:
